.travis.yml

```diff
@@ -34,21 +32,24 @@ jobs:
         # This step will run indefinitely if we do not limit it to 5 mins.
         # If the entire build reaches 50 mins, the after_failure steps never run.
         - .travis/pulp_file-tests.sh
-      deploy:
-        provider: script
-        # Push image to quay. That is our upstream "deployment".
-        script: sudo .travis/deploy.sh
-        on:
-          # Only run on the master branch, because "latest" is hardcoded
-          branch: master
-          # Also, deploy will never run on PRs.
-          # "Note that pull request builds skip the deployment step altogether."
-          # https://docs.travis-ci.com/user/deployment#pull-requests
     - name: test-insta-demo
       script:
         - sudo operator-sdk build quay.io/pulp/pulp-operator:latest
         - sudo docker images
         - sudo insta-demo/pulp-insta-demo.sh
+    - name: molecule
+      dist: xenial
+      install:
+        - pip3 install docker molecule ansible-lint yamllint flake8 openshift jmespath
+      before_script: skip
+      script: .travis/molecule.sh
+    - stage: deploy
+      name: push-to-quay
+      # Push image to quay. That is our upstream "deployment".
+      script:
+        - sudo operator-sdk build quay.io/pulp/pulp-operator:latest
+        - sudo .travis/deploy.sh
+      if: type != pull_request
```

Comment:
Wait, so when is the push-to-quay stage done? We have 3 tests now, which one(s) is it done at the end of?

I looked at the tests, and I didn't see it.

Since we still rely on the ports being changed, we should just run molecule as a separate test, but still push at the end of test-all-components (with pulp_file tests at the end.)

Reply:
deploy stage will only start if the 3 test were successful, it is similar from what we have at the plugins
https://docs.travis-ci.com/user/build-stages/#naming-your-build-stages

Reply:
molecule ports are changed in a different job/stage

Reply:
@fao89 Thanks, I was looking for it in the PR output. I read too quickly :/